packages/zevm-app-contracts/contracts/zeta-points/InvitationManager.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.7;
+
+contract InvitationManager {
+    /* An ECDSA signature. */
+    struct Signature {
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+    // Records the timestamp when a particular user gets verified.
+    mapping(address => uint256) public userVerificationTimestamps;
+
+    // Records the timestamp when a particular user accepted an invitation from an inviter.
+    mapping(address => mapping(address => uint256)) public acceptedInvitationsTimestamp;
+
+    // Store invitees for each inviter
+    mapping(address => address[]) public inviteeLists;
+
+    // Total invites accepted by day (using the start timestamp of each day as key)
+    mapping(uint256 => uint256) public totalInvitesByDay;
```

Comment:
Do we really need this? Shouldn't it be on the indexing side?

Reply:
we agree with @juansalvatore to keep some data here. the reason is there are two use cases, one is the user page, and the other is the dashboard. for the dashboard we use a graph but for the user maybe we will call directly the contracts